picard/version.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+# Copyright (C) 2019 Philipp Wolfer
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+import re
+
+
+class VersionError(Exception):
+    pass
+
+
+class Version(tuple):
```

Comment:
Why not use total_ordering to get the comparators?

Reply:
Why not use a named tuple? From typing.

Reply:
> Why not use total_ordering to get the comparators?

This would be best, but doesn't work since all comparators are already implemented (inherited from `tuple`) and `total_ordering` only fills in the missing ones. So we have to override them.